lib/db/memory.js

```diff
@@ -242,6 +242,18 @@ MemoryStore.prototype = {
     delete this.tokens[unbuf(id)];
     return P.resolve();
   },
+  /**
+   * Get all services that have have non-expired tokens
+   * @param email User's email
+   * @returns {*}
+   */
+  getActiveServicesByEmail: function getActiveServicesByEmail(email) {
```

Comment:
Correct, since the email address for a user can change, it should always be via `uid`.